src/main/java/appeng/core/sync/packets/PacketCompassResponse.java

```diff
@@ -20,59 +20,75 @@
 
 
 import io.netty.buffer.ByteBuf;
-import io.netty.buffer.Unpooled;
 
-import net.minecraft.entity.player.EntityPlayer;
+import cpw.mods.fml.common.network.simpleimpl.MessageContext;
 
 import appeng.core.sync.AppEngPacket;
-import appeng.core.sync.network.INetworkInfo;
+import appeng.core.sync.AppEngPacketHandler;
 import appeng.hooks.CompassManager;
 import appeng.hooks.CompassResult;
 
 
-public class PacketCompassResponse extends AppEngPacket
+public class PacketCompassResponse implements AppEngPacket, AppEngPacketHandler<PacketCompassResponse, AppEngPacket>
 {
 
-	public final long attunement;
-	public final int cx;
-	public final int cz;
-	public final int cdy;
-
-	public CompassResult cr;
+	private long attunement;
+	private int cx;
+	private int cz;
+	private int cdy;
+	private CompassResult cr;
+	private boolean hasResult;
+	private boolean spin;
+	private double radians;
 
 	// automatic.
-	public PacketCompassResponse( ByteBuf stream )
+	public PacketCompassResponse()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1186)